pandas/core/missing.py

```diff
@@ -314,6 +289,53 @@ def interpolate_1d(
         return result
 
 
+def _derive_indices_of_nans_to_preserve(
+    yvalues, valid, invalid, limit, limit_area, limit_direction
```

Comment:
can you try to type these * add a Paramaters / Returns section

Reply:
Yes. Will be update in next series of commits